src/controllers/restatedeployment/reconcilers/knative.rs

```diff
@@ -0,0 +1,938 @@
+    let containers = rsd
+        .spec
+        .template
+        .spec
+        .as_ref()
+        .and_then(|spec| spec.get("containers"))
+        .ok_or_else(|| Error::InvalidRestateDeployment("Missing containers in template".into()))?;
+
+    // Ensure Restate port and validate for Knative compatibility
+    let mut containers_array: Vec<serde_json::Value> = serde_json::from_value(containers.clone())?;
+    validate_knative_containers(&containers_array)?;
+    ensure_restate_port(&mut containers_array)?;
+    ensure_readiness_probe(&mut containers_array)?;
+
+    // Create owner reference
+    let owner_reference = rsd.controller_owner_ref(&()).unwrap();
+
+    Ok(json!({
+        "apiVersion": "serving.knative.dev/v1",
+        "kind": "Configuration",
+        "metadata": {
+            "name": name,
+            "namespace": namespace,
+            "ownerReferences": [owner_reference],
+            "labels": {
+                "app.kubernetes.io/managed-by": "restate-operator"
+            }
+        },
+        "spec": {
+            "template": {
+                "metadata": {
+                    "annotations": annotations,
+                    "labels": {
+                        "app.kubernetes.io/managed-by": "restate-operator"
+                    }
+                },
+                "spec": {
+                    "containers": containers_array,
+                }
+            }
+        }
+    }))
+}
+
+/// Validate containers for Knative compatibility
+/// Knative has strict requirements that differ from standard Kubernetes
+fn validate_knative_containers(containers: &[serde_json::Value]) -> Result<()> {
+    for (idx, container) in containers.iter().enumerate() {
+        // Validate port names
+        // Knative only allows port names: empty, "h2c", or "http1"
+        if let Some(ports) = container.get("ports").and_then(|p| p.as_array()) {
+            for port in ports {
+                if let Some(port_name) = port.get("name").and_then(|n| n.as_str()) {
+                    if port_name != "h2c" && port_name != "http1" {
+                        return Err(Error::InvalidRestateDeployment(
+                            format!(
+                                "Container {} has invalid port name '{}'. Knative only allows port names: 'h2c' or 'http1'. \
+                                Please update the port name in spec.template.spec.containers[{}].ports",
+                                container.get("name").and_then(|n| n.as_str()).unwrap_or("unknown"),
+                                port_name,
+                                idx
+                            )
+                        ));
+                    }
+                }
+            }
+        }
+    }
+
+    Ok(())
+}
+
+/// Ensure containers have Restate port 9080 with h2c protocol
+/// If port 9080 doesn't exist, add it. Otherwise, leave ports as-is.
+fn ensure_restate_port(containers: &mut Vec<serde_json::Value>) -> Result<()> {
```

Comment:
i don't love that users are *forced* to use 9080. is there some way that we could just assume any h2c or http1 named port is the restate port?